picard/acoustid/manager.py

```diff
@@ -62,29 +62,36 @@ def remove(self, file):
             del self._fingerprints[file]
         self._check_unsubmitted()
 
+    def is_submitted(self, file):
+        submission = self._fingerprints.get(file)
+        if submission:
+            return not submission.recordingid or submission.orig_recordingid == submission.recordingid
+        return True
+
     def _unsubmitted(self):
-        for submission in self._fingerprints.values():
+        for file, submission in self._fingerprints.items():
             if submission.recordingid and submission.orig_recordingid != submission.recordingid:
-                yield submission
+                yield (file, submission)
 
     def _check_unsubmitted(self):
         enabled = next(self._unsubmitted(), None) is not None
         self.tagger.window.enable_submit(enabled)
 
     def submit(self):
-        fingerprints = list(self._unsubmitted())
-        if not fingerprints:
+        submissions = list(self._unsubmitted())
+        if not submissions:
             self._check_unsubmitted()
             return
         log.debug("AcoustID: submitting ...")
         self.tagger.window.set_statusbar_message(
             N_('Submitting AcoustIDs ...'),
             echo=None
         )
+        fingerprints = [fingerprint for _, fingerprint in submissions]
```

Comment:
Even though it is a standard way to declare an unused variable in Python, since it is also used for gettext stuff, I would not use `_` at all.

Reply:
That's true. It is not really an issue here since this is scoped inside the list comprehension, but you are right we should avoid this.

On a similar note this specific source file makes heavy use of variables named `file`, overwriting the Python `file`. In most places I think we use `file_` for this, but we have a couple of more occurrences of just `file` I think. I will address both in a separate commit.

Reply:
Yup, the `file` one is really boring, Python guys chose a bad name for their module...